src/cascadia/TerminalApp/Pane.cpp

```diff
@@ -367,16 +368,23 @@ void Pane::_ControlWarningBellHandler(const winrt::Windows::Foundation::IInspect
     auto paneProfile = settings.FindProfile(_profile.value());
     if (paneProfile)
     {
+        if (WI_IsFlagSet(paneProfile.BellStyle(), winrt::Microsoft::Terminal::Settings::Model::BellStyle::Visual))
+        {
+            // Visual is set, raise the event with the bool set to true
+            _PaneRaiseBellHandlers(nullptr, true);
+        }
+        else
+        {
+            // Visual is not set, raise the event with the bool set to false
+            _PaneRaiseBellHandlers(nullptr, false);
+        }

```

Comment:
there's still an opportunity for simplification here :smile:

Consider the transformation...


```
if (boolean_expression) {
    doThing(true);
} else {
    doThing(false);
}
```

Reply:
okay _now_ its fixed